src/Logger/Logger.cc

```diff
@@ -6,15 +6,15 @@
 
 #include "Logger.h"
 
+#include "Main/ProgramSettings.h"
+
 #include <string>
 
 namespace carta {
 namespace logger {
 
-static bool log_protocol_messages(false);
-
-void InitLogger(bool no_log_file, int verbosity, bool log_performance, bool log_protocol_messages_, fs::path user_directory) {
-    log_protocol_messages = log_protocol_messages_;
+void InitLogger() {
+    auto& settings = ProgramSettings::GetInstance();
```

Comment:
In this function, after this line, set the static variable `log_protocol_messages` that was there in the original code (copying the value from this settings object), so that the other functions (which only use that one setting value) can use this variable instead of fetching the whole settings object.